widgets/url-sync/index.js

```diff
@@ -0,0 +1,172 @@
+'use strict';
+
+var AlgoliaSearchHelper = require('algoliasearch-helper').AlgoliaSearchHelper;
+
+var isEqual = require('lodash/lang/isEqual');
+var merge = require('lodash/object/merge');
+var forEach = require('lodash/collection/forEach');
+
+var getNow = Date.now ? function() { return Date.now(); } : function() { return (new Date()).getTime(); };
+
+var timerMaker = function(t0){
+  var t = t0;
+  return function timer(){
+    var now = getNow();
+    var delta = now - t;
+    t = now;
+    return delta;
+  };
+};
+
+/**
+ * @typedef {object} UrlUtil
+ * @property {string} character the character used in the url
+ * @property {function} onpopstate add an event listener for the URL change
+ * @property {function} pushState creates a new entry in the browser history
+ * @property {function} replaceState update the current entry of the browser history
+ * @property {function} readUrl reads the query string of the parameters
+ */
+
+/**
+ * Handles the legacy browsers
+ * @type {UrlUtil}
+ */
+var hashUrlUtils = {
+  character: '#',
+  onpopstate: function(cb){
+    if (window.addEventListener) {
```

Comment:
Yes you can rely on the polyfill for this https://github.com/Financial-Times/polyfill-service/blob/master/polyfills/Event/polyfill-ie8.js#L88